### PR TITLE
[add] added support for FT.SPELLCHECK

### PR DESCRIPTION
**Pull request by filipecosta90**
added support for FT.SPELLCHECK.

the optional parameters:
- inclusion named dictionary
- exclusion named dictionary
- distance 
are all supported with the addition of SpellCheckOptions struct. 
-------
Regarding the command response handling, the method SpellCheck with signature:

`func (i *Client) SpellCheck(q *Query, s *SpellCheckOptions) (suggs []MisspelledTerm, total int, err error)`

returns:
-  suggestions for misspelled terms in the format of a slice of MisspelledTerm (contains the misspelled term and a sortable list of suggestions)
-  the total number of results
- or an error if something went wrong

### CODE REVIEW

File: redisearch/client.go [hunk 1]
```diff
@@ -346,6 +346,49 @@ func (i *Client) Search(q *Query) (docs []Document, total int, err error) {
...
+			return
+		}
+
+		if d, e := loadMisspelledTerm(termArray, termIdx, suggIdx); e == nil {
+			suggs = append(suggs, d)
+			if d.Len() > 0 {
+				total++
+			}
+		} else {
+			log.Print("Error parsing misspelled suggestion: ", e)
```
Comment (MeirShpilraien) [inline redisearch/client.go]: Don't we want to return an error here?

File: redisearch/spellcheck.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	}
+	missT = NewMisspelledTerm(term)
+	lst, err := redis.Values(arr[suggIdx], err)
+	if err != nil {
+		return MisspelledTerm{}, fmt.Errorf("Could not the array of suggestions for spelling corrections on term %s. Error: %s", term, err)
+	}
+	for i := 0; i < len(lst); i++ {
+		innerLst, err := redis.Values(lst[i], err)
+		if err != nil {
+			return MisspelledTerm{}, fmt.Errorf("Could not the inner array of suggestions for spelling corrections on term %s. Error: %s", term, err)
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply redisearch/spellcheck.go]: @MeirShpilraien please revise latest commit

File: redisearch/spellcheck.go [hunk 3]
```diff
@@ -113,20 +113,20 @@ func loadMisspelledTerm(arr []interface{}, termIdx, suggIdx int) (missT Misspell
 	missT = NewMisspelledTerm(term)
 	lst, err := redis.Values(arr[suggIdx], err)
 	if err != nil {
-		return MisspelledTerm{}, fmt.Errorf("Could not the array of suggestions for spelling corrections on term %s. Error: %s", term, err)
+		return MisspelledTerm{}, fmt.Errorf("Could not ge the array of suggestions for spelling corrections on term %s. Error: %s", term, err)
```
Comment (MeirShpilraien) [inline redisearch/spellcheck.go]: ge -> get :)

File: redisearch/spellcheck.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+		return fmt.Errorf("The maximal Levenshtein distance for spelling suggestions should be between [1,4]. Got %d", distance)
+	} else {
+		s.Distance = distance
+	}
+	return nil
+}
+
+// AddExclusionDict adds a custom dictionary named {dictname} to the exclusion list
+func (s *SpellCheckOptions) AddExclusionDict(dictname string) {
+	s.ExclusionDicts = append(s.ExclusionDicts, dictname)
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply redisearch/spellcheck.go]: tks @ericfengchao. Proceeded with the requested changes

**Pull request merged**